### PR TITLE
Notebook Code Cleanup + Utils Test

### PR DESCRIPTION
**Pull request by chlafreniere**
More cleanup (JupyterNotebookProvider handles cleanup, not the controller... Which was always trying to cleanup an empty array!)

Also added to a couple of simple tests.

### CODE REVIEW

File: extensions/notebook/src/test/common/utils.test.ts [hunk 1]
```diff
@@ -47,10 +47,16 @@ describe('Utils Tests', function () {
 
 	it('getOSPlatform', async () => {
 		should(utils.getOSPlatform()).not.throw();
+		should(utils.getOSPlatform('win32')).equal(utils.Platform.Windows);
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/test/common/utils.test.ts:50]: So I'm pretty strongly against this type of testing. Hardcoding platform checks in tests just to get code coverage isn't something we should be doing - instead we should be making the tests be able to run on any platform and then actually running the tests on all platforms.

Now obviously we aren't there yet for running on all platforms but that I don't see that as a good reason to do this kind of thing now.

File: extensions/notebook/src/common/utils.ts [hunk 2]
```diff
@@ -121,8 +121,8 @@ export interface IEndpoint {
 	protocol: string;
 }
 
-export function getOSPlatform(): Platform {
-	switch (process.platform) {
+export function getOSPlatform(platform?: NodeJS.Platform): Platform {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/common/utils.ts:124]: Honestly I'd just remove this method completely. This is just adding an extra layer for no real benefit - the return NodeJS.Platform is already a set of known values so adding our own mapping on top of it seems a bit unnecessary.

Comment (chlafreniere) [reply extensions/notebook/src/common/utils.ts:124]: That's fine. Let me do that. We have to keep getOSPlatformId() for now (it's how our python packages are named) but lemme rewrite some code to remove util method.

File: extensions/notebook/src/test/common/utils.test.ts (hunk 1 above)
Comment (chlafreniere) [reply extensions/notebook/src/test/common/utils.test.ts:50]: Okayyy

File: extensions/notebook/src/common/utils.ts (hunk 2 above)
Comment (Charles-Gagnon) [reply extensions/notebook/src/common/utils.ts]: Yup, that part is fine. It'd just switch on process.platform

File: extensions/notebook/src/book/remoteBookController.ts [hunk 3]
```diff
@@ -73,7 +73,7 @@ export class RemoteBookController {
 	public async getAssets(release?: IRelease): Promise<IAsset[]> {
 		if (release) {
 			let format: string[] = [];
-			if (utils.getOSPlatform() === utils.Platform.Windows || utils.getOSPlatform() === utils.Platform.Mac) {
+			if (process.platform === winPlatform || process.platform === macPlatform) {
```
Comment (Charles-Gagnon) [inline extensions/notebook/src/book/remoteBookController.ts:76]: Do you need the constants elsewhere for something? If they're just being used to compare you can just use the string values - process.platform is a string literal union so the type checking will already guarantee you don't mistype it or something

Comment (chlafreniere) [reply extensions/notebook/src/book/remoteBookController.ts:76]: Looks like just for comparison to what process.platform returns

**Pull request merged**